### PR TITLE
Add devcontainer

### LINKED ISSUE #331: Add vscode dev-container
It would be much easier to contribute to the project if there is a VSCode Dev-Container definition in this project. This allows working on this project without installing any tools (beside vscode and docker).

### PR DESCRIPTION
**Pull request by pixelrazor**
Fixes #331 
The dockerfile could probably be optimized - i just tinkered until i got something that wants to work out the box. the cargo install step is pretty hefty, so i COULD try to tinker with making that a separate layer as early as possible to help cache it, but i don't think it's too important given this shouldn't change often at all.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM rust:1.66
+
+RUN apt update && \
+    curl -fsSL https://deb.nodesource.com/setup_lts.x | bash - && \
+    apt update && \
+    apt install -y --no-install-recommends nodejs libssl-dev musl-dev make perl curl 
+
+RUN RUSTFLAGS=-Ctarget-feature=-crt-static cargo install wasm-pack cargo-chef \
```
Comment (nitnelave) [inline .devcontainer/Dockerfile:8]: Either you don't need cargo-chef, or (better) you can steal the dependency pre-compilation step from this dockerfile: https://github.com/nitnelave/lldap/blob/main/Dockerfile#L25-L41

Comment (pixelrazor) [reply .devcontainer/Dockerfile:8]: I'll opt to remove chef. The container itself doesn't contain the code - vscode bind mounts it after creation

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    && npm install -g rollup \
+    && rustup target add wasm32-unknown-unknown
```
Comment (pixelrazor) [inline .devcontainer/Dockerfile:10]: I'm going to add auser in the container - vscode will amp that user's UID and GID to the current user and ffix the file permission issues

**Comment by nitnelave** [comment]
Thanks, that looks better! I'm no expert on VS code containers, so I'll trust you that it works.

Last nit: can you add `.devcontainer` to the `.dockerignore` file?

**Comment by pixelrazor** [comment]
> Thanks, that looks better! I'm no expert on VS code containers, so I'll trust you that it works.
> 
> Last nit: can you add `.devcontainer` to the `.dockerignore` file?

Done. I added `.vscode` to gitignore as well - that wolds workspace specific configuration for vscode

**Pull request merged**